.github/actions/save_cache/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Save cache
+description: Save conan and ccache cache for develop branch
+inputs:
+  conan_dir:
+    description: Path to .conan directory
+    required: true
+  ccache_dir:
+    description: Path to .ccache directory
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: Find common commit
+      id: git_common_ancestor
+      uses: ./.github/actions/git_common_ancestor
+
+    - name: Cleanup conan directory from extra data
+      shell: bash
+      run: |
+        conan remove "*" -s -b -f
+
+    - name: Save conan cache
+      uses: actions/cache/save@v3
+      with:
+        path: ${{ inputs.conan_dir }}/data
+        key: clio-conan_data-${{ runner.os }}-develop-${{ hashFiles('./conanfile.py') }}
```

Comment:
i am thinking if there is any chance , the runner.os does not change and conanfile.py does not change, but the cache is outdated? for example, the compiler gets updated? 
Do we need to also consider to add build mode (debug/release)?

Reply:
We can always manually delete cache here: https://github.com/XRPLF/clio/actions/caches
We have only release builds now, we could change key later if needed later. Cache miss costs only about 1 hour of CI work

Reply:
The Conan packages' hashes encapsulate the compiler version so that should invalidate the cache right?

Reply:
If we update compiler, conan will not use packages in local cache. But with the current implementation because `conanfile.py` didn't change, CI will not upload a new cache to the github storage.

I found a better solution:
```bash
conan info . -j info.json
cat info.json | jq '.[] | "\(.id), \(.package_revision)"' | shasum -a 256 -
```
This will take id and revision from every conan package used and calculate sha256 from it. So if we change compiler packages revisions will be different and the sha256 hash will be different.